modules/react/action-bar/lib/ActionBar.tsx

```diff
@@ -1,37 +1,39 @@
 import * as React from 'react';
-import {createComponent, styled, StyledType} from '@workday/canvas-kit-react/common';
-import {commonColors, colors, space} from '@workday/canvas-kit-react/tokens';
-import {HStack, HStackProps, StackSpacing} from '@workday/canvas-kit-react/layout';
+import {createContainer} from '@workday/canvas-kit-react/common';
+import {Menu} from '@workday/canvas-kit-react/menu';
+import {useActionBarModel} from './useActionBarModel';
+import {ActionBarList} from './ActionBarList';
+import {ActionBarOverflowButton} from './ActionBarOverflowButton';
+import {ActionBarItem} from './ActionBarItem';
 
-interface ActionBarProps extends Omit<HStackProps, 'spacing'> {
-  spacing?: StackSpacing;
+export interface ActionBarProps {
+  /**
+   * The contents of the ActionBar. Can be `ActionBar` children or any valid elements.
+   */
+  children: React.ReactNode;
 }
 
-const ResponsiveHStack = styled(HStack)<HStackProps & StyledType>(({theme}) => ({
-  [theme.canvas.breakpoints.down('s')]: {
-    padding: space.s,
-    '> *:not(div)': {
-      flex: 1,
-    },
-  },
-}));
-
-export const ActionBar = createComponent('div')({
+export const ActionBar = createContainer('div')({
   displayName: 'ActionBar',
-  Component: (elemProps: ActionBarProps, ref, Element) => (
-    <ResponsiveHStack
-      ref={ref}
-      as={Element}
-      spacing="s"
-      depth={1}
-      background={commonColors.background}
-      borderTop={`solid 1px ${colors.soap400}`}
-      padding={`${space.s} ${space.xl} `}
-      position="fixed"
-      bottom={0}
-      left={0}
-      right={0}
-      {...elemProps}
-    />
-  ),
+  modelHook: useActionBarModel,
+  subComponents: {
+    List: ActionBarList,
+    Item: ActionBarItem,
```

Comment:
I could go either way, but `item` is always a `button` maybe this was following tabs convention but should this we renamed to `ActionBar.Button` ?

```tsx
  <ActionBar>
     <ActionBar.List>
         <ActionBar.Button isPrimaryButton />
         <ActionBar.Button />
     </ActionBar.List>
   </ActionBar>
```

I'm just thinking out loud. Maybe `Item` makes sense because you're wrapping it in a `List`

Reply:
this isn't a blocker BTW

Reply:
Yeah, `ActionBar.Item` have more sense to be used here, because it's a part of the list and also refers to action, not jut button. Also `Item` is closely connected with menu, so it's shows that `ActionBar.Item` is the same with items in the overflow menu.

Reply:
An also it refers to model state items

Reply:
ok cool! make sense to me!